crates/node/src/network_server/handler/cluster_ctrl.rs

```diff
@@ -45,14 +62,86 @@ impl ClusterCtrlSvc for ClusterCtrlSvcHandler {
             .controller_handle
             .get_cluster_state()
             .await
-            .map_err(|_| tonic::Status::aborted("Node is shutting down"))?;
+            .map_err(|_| Status::aborted("Node is shutting down"))?;
 
         let resp = ClusterStateResponse {
             cluster_state: Some((*cluster_state).clone().into()),
         };
         Ok(Response::new(resp))
     }
 
+    async fn list_logs(
+        &self,
+        _request: Request<ListLogsRequest>,
+    ) -> Result<Response<ListLogsResponse>, Status> {
+        Ok(Response::new(ListLogsResponse {
+            data: serialize_value(self.logs().await?),
```

Comment:
Tiny nit: It's a little unconventional style to have async function names without a verb. Often functions with no prefixing verb behave like properties/getters. In your case, it's a getter, it's just a little unconventional for "async" getters to be async in this style.

To be clear, I'm not asking you to change, but it's more a point to help us come to internal conventions and style.

Reply:
Good callout! Noted on convention, and renamed. I agree, it does look a bit odd in hindsight.